### PR TITLE
Computing distances between two empty PersistenceDiagrams

### PR DESCRIPTION
**Pull request by mbudisic**
A distance (either Wasserstein or Bottleneck) between two empty PersistenceDiagrams should return zero. (Empty diagrams contain trivially the "diagonal" in the birth-death graph, and distance between the diagram and itself should be zero.)

Before this change, the code produces an error when applying either `hungarian` or `_hopcroft_karp` algorithms in `matching.jl`, and the two errors are different.

The pull request performs a simple test at the beginning of `Bottleneck()` and `Wasserstein()` functions and returns the distance zero or a zero `Matching`.

The pull additionally adds a unit test group for this issue.

### CODE REVIEW

File: src/matching.jl [hunk 1]
```diff
@@ -402,6 +402,14 @@ function (::Bottleneck)(left::PersistenceDiagram, right::PersistenceDiagram; mat
         end
     end
 
+    if length(left) == 0 & length(right) == 0 
+        if matching
+            return Matching(left, right, 0, Pair{Int,Int}[], true)
+        else
+            return 0
```
Comment (mtsch) [inline src/matching.jl:409]:
```suggestion
            return 0.0
```

File: src/matching.jl [hunk 2]
```diff
@@ -494,6 +502,15 @@ end
 function (w::Wasserstein)(
     left::PersistenceDiagram, right::PersistenceDiagram; matching=false
 )
+
+    if length(left) == 0 & length(right) == 0 
+        if matching
+            return Matching(left, right, 0, Pair{Int,Int}[], false)
+        else
+            return 0
```
Comment (mtsch) [inline src/matching.jl:510]:
```suggestion
            return 0.0
```

File: test/matching.jl [hunk 3]
```diff
@@ -79,6 +79,14 @@ end
     @test weight(Wasserstein(), diag1, diag1) ≡ 0.0
 end
 
+@testset "Empty diagrams" begin
+    diag = PersistenceDiagram([])
+
+    @test Wasserstein()(diag,diag; matching = false) ≡ 0
+    @test Bottleneck()(diag,diag; matching = false) ≡ 0
```
Comment (mtsch) [inline test/matching.jl:86]:
```suggestion
    @test Wasserstein()(diag,diag; matching = false) ≡ 0.0
    @test Bottleneck()(diag,diag; matching = false) ≡ 0.0
```

**Review by mtsch — CHANGES_REQUESTED** [review]
Thanks so much, I can't believe I missed such an obvious edge case!

Looks good, but I'd rather it returns a `0.0`, so the distances are always `Float64`s.

**Comment by mbudisic** [comment]
Ah yes - I didn't pay attention to this. I accepted the suggestions.

Thanks for developing the package, my collaborators and I are enjoying using it!

**Review by mtsch — APPROVED** [review]
Thanks! Looks good now. I'll bump the version and merge/register it now.

**Comment by mtsch** [comment]
The fix is registered and can now be installed with the package manager. Thanks again!

**Comment by mbudisic** [comment]
My pleasure!
